src/upload/UploadList.stories.tsx

```diff
@@ -20,26 +19,69 @@ function UploadListWithWrapper(props: Partial<UploadList_Props_t>) {
         progress: 0.5,
     }] as IUploadedFile[]);
 
-    return <UploadList
-        onUploadCancel={(file: IUploadedFile) => {
-            setFiles((existingFiles) => existingFiles.filter((f) => f !== file));
-            action("onUploadCancel")([file]);
-        }}
-        onUploadClick={action("uploadClick")}
-        files={files}
-        placeholder={<div style={{ height: 100 }}>Drop your files here to upload</div>}
-        {...props}
-    />;
+    return <div id="uploadListDiv">
+        <UploadList
+            onUploadCancel={(file: IUploadedFile) => {
+                setFiles((existingFiles) => existingFiles.filter((f) => f !== file));
+                action("onUploadCancel")([file]);
+            }}
+            onUploadClick={action("uploadClick")}
+            files={files}
+            placeholder={<div style={{ height: 100 }}>Drop your files here to upload</div>}
+            {...props}
+        />
+    </div>;
 }
 
 export const normal = () => <Paper>
     <UploadListWithWrapper />
 </Paper>;
 
+normal.story = {
+    parameters: {
+        async puppeteerTest(page: any) {
+            const uploadClickActionPromise = interceptAction(page, "uploadClick");
+
+            const uploadListItemsSelector = "div#uploadListDiv > div > *";
+            let uploadListItems = await page.$$(uploadListItemsSelector);
+
+            expect(uploadListItems.length).toBe(2);
+            await uploadListItems[0].click();
+            const uploadClickActionData = await uploadClickActionPromise;
+            expect(uploadClickActionData.name).toBe("uploadClick");
+            expect(uploadClickActionData.args[0]).toEqual({ fileName: "Filename.txt", progress: 1 });
+
+            const uploadCancelActionPromise = interceptAction(page, "onUploadCancel");
+            const button = await page.$("button");
+            await button.click();
+            const uploadCancelActionData = await uploadCancelActionPromise;
+            expect(uploadCancelActionData.name).toBe("onUploadCancel");
+            expect(uploadCancelActionData.args[0][0]).toEqual({ fileName: "File2.doc", progress: 0.5 });
+            uploadListItems = await page.$$(uploadListItemsSelector);
+            expect(uploadListItems.length).toBe(1);
+            const itemName = await uploadListItems[0].evaluate((item: HTMLDivElement) => item.innerText);
+            expect(itemName).toContain("Filename.txt");
+
+            stopIntercept(page);
+        },
+    },
+};
+
 export const empty = () => <Paper>
     <UploadListWithWrapper files={[]} />
 </Paper>;
 
+empty.story = {
```

Comment:
No need to test for this.
Screenshot tests will already catch when something visual changes.